### PR TITLE
WIP: Implement more continuous log CDF functions

### PR DESCRIPTION
**Pull request by domenzain**
Implement several log CDF functions

The distributions concerned are:
- [x] Pareto
- [x] Flat
- [x] Wald
- [x] Exponential

Each implementation is then tested against SciPy's own implementation.

### CODE REVIEW

**Comment by domenzain** [comment]
This is done in the context of  #1867 and #1864.
When it is complete, there should be an adequate CDF for all continuous distributions.

File: pymc3/tests/test_distributions.py
Comment (springcoil) [inline pymc3/tests/test_distributions.py]: I would check infinite values. Seems a good check.

**Review by springcoil — COMMENTED** [review]
Looks cool to me. Good contributions. If you could add a few tests for infinite values that'd be cool. Otherwise we can leave it for another commit.

**Comment by springcoil** [comment]
I guess we need to wait for Beta to get this to merge. But good work @domenzain :)

File: pymc3/distributions/continuous.py
Comment (domenzain) [inline pymc3/distributions/continuous.py]: I don't know where to put this. I'm guessing there must be a more logical place than here.

File: pymc3/tests/test_distributions.py
Comment (domenzain) [reply pymc3/tests/test_distributions.py]: I've pushed a quick explicit couple of tests.

File: pymc3/distributions/continuous.py
Comment (fonnesbeck) [inline pymc3/distributions/continuous.py]: use increment `value += alpha`

File: pymc3/distributions/continuous.py
Comment (ferrine) [inline pymc3/distributions/continuous.py]: You can move it to `dist_math`

File: pymc3/distributions/continuous.py
Comment (ferrine) [inline pymc3/distributions/continuous.py]: What if value is tensor? Mb `tt.as_tensor(value)` fits better there

File: pymc3/distributions/continuous.py
Comment (ferrine) [inline pymc3/distributions/continuous.py:191]: Do you check vector value for this op?

File: pymc3/distributions/continuous.py
Comment (ferrine) [inline pymc3/distributions/continuous.py]: Does it work with ndimentional tensors?

**Review by ferrine — CHANGES_REQUESTED** [review]
Does it all work with tensors, not scalars, we need more tests for that

File: pymc3/distributions/continuous.py
Comment (domenzain) [reply pymc3/distributions/continuous.py:191]: I think all of these are element-wise operations.
When a tensor is passed as argument they would behave as expected:

```python
>>> tt.eq(np.array([1, 2, np.nan, -np.inf, np.inf]), np.inf).eval()
array([False, False, False, False,  True], dtype=bool)
```

File: pymc3/distributions/continuous.py
Comment (domenzain) [reply pymc3/distributions/continuous.py]: Done.

File: pymc3/distributions/continuous.py
Comment (domenzain) [reply pymc3/distributions/continuous.py]: I'm unsure about broadcasting when the tensors will not be shaped identically.
But otherwise everything is element-wise ok.

File: pymc3/distributions/continuous.py
Comment (domenzain) [reply pymc3/distributions/continuous.py]: Changing this. I'm using `floatX` and `as_tensor` instead.

**Review by domenzain — COMMENTED** [review]
Latest review taken into account and integrated in commits where applicable.

**Comment by domenzain** [comment]
I've removed `Beta` from this pull request, seeing that @fonnesbeck is working on it directly on the target branch.

**Comment by fonnesbeck** [comment]
@domenzain I was giving it a try, but have not finished yet.

Keep in mind, everyone, that this is being merged into a feature branch, so there will be another opportunity to review it before it is merged into master.

**Comment by fonnesbeck** [comment]
@domenzain you may need to rebase your branch.

**Comment by fonnesbeck** [comment]
I will merge this into my feature branch.

**Pull request merged**
